stdlib/public/core/StringGuts.swift

```diff
@@ -1127,3 +1010,141 @@ extension _StringGuts {
     }
   }
 }
+
+extension _StringGuts : Sequence {
```

Comment:
Why do we need to make _StringGuts be a `Sequence`? I was hoping we could avoid all conformances like this to stave off the temptation for premature generalization and using _StringGuts in generic contexts.

Reply:
The conformance is completely unnecessary; I'll remove it.

The `Iterator` made it easier to get `UTF8View` off `_StringCore`. We may or may not decide to keep it, although a single buffering iterator is one way to remove those ugly distinct iterator properties from `UnicodeScalarView`...